src/libraries/System.Net.WebSockets/src/System/Net/WebSockets/AsyncMutex.cs

```diff
@@ -185,6 +188,8 @@ public void Exit()
 
             void Contended()
             {
+                if (NetEventSource.Log.IsEnabled()) NetEventSource.MutexContended(this, _gate);
```

Comment:
I am not sure I understand your comment correctly (I have several options 😂😅) Do you mean that this method should never be called, or that the raised event meaning doesn't exactly align with what the method means?

Reply:
What is the purpose of this event? It's the exact same one being raised for enter, but they're two completely different things. Beyond the fact that one is about entering the lock and the other is about exiting, exiting is _never_ actually contended, because the only thread calling this is the sole owner of the mutex and the only one allowed to call Exit (another thread calling it would be programmer error). The name of the method here being Contended is about the state of the lock, not about this particular access. So, using the same MutexContended that's used by Enter just muddies whatever purpose this event is meant to serve.

Reply:
https://github.com/dotnet/runtime/pull/107662#discussion_r1752918045